lib/Sema/TypeCheckStmt.cpp

```diff
@@ -1990,9 +1990,16 @@ static void checkClassConstructorBody(ClassDecl *classDecl,
 
   // A class designated initializer must never be delegating.
   if (ctor->isDesignatedInit() && isDelegating) {
-    ctor->diagnose(diag::delegating_designated_init,
+
+    if (classDecl->getForeignClassKind() == ClassDecl::ForeignKind::CFType) {
+        ctx.Diags.diagnose(ctor->getLoc(), diag::delegating_designated_init_in_extension,
```

Comment:
I think you can just do `ctor->diagnose(...)` like you did in the else branch.

Reply:
yep, was much more simple! thanks!